examples/unix-domain-socket/src/main.rs

```diff
@@ -118,22 +92,14 @@ mod unix {
         peer_cred: UCred,
     }
 
-    impl connect_info::Connected<&UnixStream> for UdsConnectInfo {
-        fn connect_info(target: &UnixStream) -> Self {
-            let peer_addr = target.peer_addr().unwrap();
-            let peer_cred = target.peer_cred().unwrap();
-
+    impl connect_info::Connected<IncomingStream<'_, UnixListener>> for UdsConnectInfo {
```

Comment:
Is there some reason something like this can't be part of axum? I think we could implement at least `impl Connected<unix::SocketAddr> for unix::SocketAddr`, `impl Connected<IncomingStream<'_, UnixListener>> for unix::SocketAddr` and `impl Connected<IncomingStream<'_, UnixListener>> for UCreds`, right?

Reply:
You cannot do `impl Connected<unix::SocketAddr> for unix::SocketAddr` because `unix::SocketAddr` is not `Clone`.

But we could have some `UdsConnectInfo` in axum that uses `Arc<unix::SocketAddr>` (similarly to [what tonic has](https://docs.rs/tonic/latest/tonic/transport/server/struct.UdsConnectInfo.html)) but I've never actually used unix streams myself so I'm not sure what info that type should contain. Do you have an opinion about that?

Reply:
Hm, no. I've used UDS w/ axum once, but didn't care about connection info then.

Reply:
In that case I think we should just leave it as is. We can always add it if someone requests it and has a use case.